### PR TITLE
[nxp] Fix recvmsg duplicate definition on latest Zephyr tree

### LINKED ISSUE #31488: [Zephyr] Duplicate definition of recvmsg on latest Zephyr
### Reproduction steps

Build any Zephyr platform on Zephyr top of tree (example: 4d854a193e4958659669afadad621f86636b0bff)

### Platform

nrf connect, other

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

When building a Zephyr platform on Zephyr top of tree, we get a duplicate definition of recvmsg.
recvmsg was implemented recently on Zephyr side:
https://github.com/zephyrproject-rtos/zephyr/pull/65694

To mitigate compatibility between platforms running on older zephyr version (release tag for example) and development on latest Zephyr, we could put the custom recvmsg implementation from src/inet/ZephyrSocket.h under the flag NET_CONTEXT_RECV_PKTINFO, as it was created at the same time.
When everyone has migrated to next Zephyr release (3.6.0), we could completely remove the custom implementation.

I can submit a PR with this change, any feedback from other zephyr platforms (nrf, telink) is welcome.

### PR DESCRIPTION
**Pull request by axelnxp**
Fixes #31488

recvmsg was implemented recently in Zephyr:
https://github.com/zephyrproject-rtos/zephyr/pull/65694

This commit aims to mitigate this issue and keep backward compatibility by using NET_CONTEXT_RECV_PKTINFO Kconfig to differentiate Zephyr versions.

### CODE REVIEW

File: src/inet/ZephyrSocket.h [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #include <zephyr/net/socket.h>
 #endif
 
+#ifndef CONFIG_NET_CONTEXT_RECV_PKTINFO
```
Comment (Damian-Nordic) [inline src/inet/ZephyrSocket.h:34]: You can just disable `CHIP_SYSTEM_CONFIG_USE_ZEPHYR_SOCKET_EXTENSIONS` for your platform if you don't need this extension. We will probably get rid off this file when nRF and Telink update Zephyr.

**Review by Damian-Nordic — APPROVED** [review]

File: src/inet/ZephyrSocket.h (hunk 1 above)
Comment (axelnxp) [reply src/inet/ZephyrSocket.h:34]: I didn't catch this flag, thanks for pointing it out !
I'll try that instead, and decline if that works.

Comment (axelnxp) [reply src/inet/ZephyrSocket.h:34]: This solution works for us, I'm closing the PR.

**Comment by axelnxp** [comment]
The solution suggested by @Damian-Nordic makes more sense and works for us, so I'm declining the PR and closing the related issue ticket.

**Pull request closed without merging**